std/experimental/allocator/gc_allocator.d

```diff
@@ -21,7 +21,7 @@ struct GCAllocator
     deallocate) and $(D reallocate) methods are $(D @system) because they may
     move memory around, leaving dangling pointers in user code.
     */
-    @trusted void[] allocate(size_t bytes) shared
+    pure nothrow @trusted void[] allocate(size_t bytes) shared const
```

Comment:
So that code can be `pure` with the `GCAllocator`, but to be honest I wasn't really sure about this one. Note that #4228 also tries to solve a part of this problem.